### PR TITLE
Day4

### PR DESCRIPTION
**Pull request by amrendra-rathore**
## Task 1
Create a file validation.js inside the utils folder.

Define a variable 'users' of the following format.

[
{
traineeEmail: 'trainee1@successive.tech',
reviewerEmail: 'reviewer1@successive.tech',
}
]

Create a function validateEmail with the following:
- Accepts email as an argument.
- Uses a regex to validate the email and a specific domain.
- Return true if an email is correct otherwise false.

Create a function 'validateUsers' with the following:
- Accepts above-defined users as an argument.
- Iterate over the array of users.
- Validate traineeEmail and reviewerEmail using above-defined function.
- Prints all valid users with their count.
- Prints all invalid users with their count.

Acceptance Criteria:
- Should print correct no of valid and invalid users.
- Should use array methods to iterate over an array.
- Should use destructuring for fetching object properties.

## Task 2

Create a file permissions.js inside the utils folder.

Define variable named permissions in the following format:

{
'getUsers': {
all: ['head-trainer'],
read : ['trainee', 'trainer'],
write : ['trainer'],
delete: [],
}
}

Create a function named hasPermission with following:

Parameters: - moduleName i.e 'getUsers',
- role i.e 'trainee',
- permissionType i.e 'read', 'write' or 'delete' etc.

Output: Return true or false based on the parameters passed by checking in the module name, role, and permission type from the permissions object.
Proper logging inside functions.

### CODE REVIEW

File: utils/permissions.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  'getUsers': {
+    all: ['head-trainer'],
+    read: ['trainee', 'trainer'],
+    write: ['trainer'],
+    delete: [],
+  },
+};
+
+function hasPermission(moduleName, role, permissionType) {
+
```
Comment (mohit-nx) [inline utils/permissions.js]: print false if the module doesn't exists

File: utils/validation.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  }
+  else {
+    return false;
+  }
+};
+
+// Function to validate Users in the above given array of objects.
+function validateUsers() {
+  var invalid = 0, valid = 0;
+  users.map((x) => {
```
Comment (mohit-nx) [inline utils/validation.js]: use forEach

**Review by mohit-nx — CHANGES_REQUESTED** [review]

File: utils/permissions.js (hunk 1 above)
Comment (amrendra-rathore) [reply utils/permissions.js]: Added another if-else statement to check if the module exists.

File: utils/validation.js (hunk 2 above)
Comment (amrendra-rathore) [reply utils/validation.js]: used forEach instead of map.

**Review by mohit-nx — APPROVED** [review]

**Pull request merged**